### ISSUE TITLE
Incorrect type display for CAST

### ISSUE THREAD

**Issue by yyy1000** [apache/datafusion#10274]
### Describe the bug

The type for CAST is not correct when using datafusion-cli

### To Reproduce

DataFusion CLI v37.1.0
> SELECT CAST(2 AS INTEGER);
+----------+
| Int64(2) |
+----------+
| 2        |
+----------+
1 row(s) fetched. 
Elapsed 0.027 seconds.

> SELECT CAST(2 AS DOUBLE);
+----------+
| Int64(2) |
+----------+
| 2.0      |
+----------+
1 row(s) fetched. 
Elapsed 0.005 seconds.

> SELECT arrow_typeof(CAST(2 AS DOUBLE));
+------------------------+
| arrow_typeof(Int64(2)) |
+------------------------+
| Float64                |
+------------------------+
1 row(s) fetched. 
Elapsed 0.008 seconds.

### Additional context

I'm finding what's wrong with it. I think it's not an issue from datafusion-cli but from the CAST itself. 🤔

**Comment by jayzhan211** [comment]
The result seems correct for me.

+----------+
| Int64(2) |
+----------+
| 2.0 |
+----------+


i64(2) is the name, not the type.

**Comment by yyy1000** [comment]
> i64(2) is the name, not the type.

Yes, I realized that. I was thinking whether we should use a better name to display, for example `CAST(2 AS INTEGER)` in the case but not `Int64(2)` , like it in duckdb. What do you think? @jayzhan211 
```
SELECT CAST(2 as DOUBLE);
┌───────────────────┐
│ CAST(2 AS DOUBLE) │
│      double       │
├───────────────────┤
│               2.0 │
└───────────────────┘
```

**Comment by jayzhan211** [comment]
Strongly agree with better name!

**Comment by jayzhan211** [comment]
> > Yes, I realized that. I was thinking whether we should use a better name to display, for example CAST(2 AS INTEGER) in the case but not Int64(2) , like it in duckdb. What do you think? @jayzhan211
> 
> I agree formatting `Int64(2)` as `2` would certainly be nicer
> 
> I am not sure about removing the `CAST`

I think you are mistaken

Current display is like, which is unclear why i64 type has a result like float, because we do not show the `casting`
```
+----------+
| Int64(2) |
+----------+
| 2.0 |
+----------+
```

Expect display is something like
```
+----------+
| Cast(Int64(2) as Float)|
+----------+
| 2.0 |
+----------+
```

**Comment by alamb** [comment]
> Expect display is something like

I agree @jayzhan211  -- sorry for misreading this.  I agree that If the user ran
```sql
select CAST(2 AS Float)`
```

seeing this makes sense 
```
+----------+
| Cast(Int64(2) as Float)|
+----------+
| 2.0 |
+----------+
```

However, if the user wrote something like this
```sql
select col = 2
```

I would expect the result to be something like (even if col was a float column and coercsion applied casts)

```
+----------+
| col = Int64(2)|
+----------+
| 2.0 |
+----------+
```

**Comment by jayzhan211** [comment]
> > Expect display is something like
> 
> I agree @jayzhan211 -- sorry for misreading this. I agree that If the user ran
> 
> ```sql
> select CAST(2 AS Float)`
> ```
> 
> seeing this makes sense
> 
> ```
> +----------+
> | Cast(Int64(2) as Float)|
> +----------+
> | 2.0 |
> +----------+
> ```
> 
> However, if the user wrote something like this
> 
> ```sql
> select col = 2
> ```
> 
> I would expect the result to be something like (even if col was a float column and coercsion applied casts)
> 
> ```
> +----------+
> | col = Int64(2)|
> +----------+
> | 2.0 |
> +----------+
> ```

It looks good to me because we can infer the type from `col`.

**Comment by jayzhan211** [comment]
Note: I would like to know if it is possible to have one single function name for `logical` and `physical` expression. It is also quite related to the issue here.

**Comment by jayzhan211** [comment]
@yyy1000 Are you still working on this?

**Comment by yyy1000** [comment]
> @yyy1000 Are you still working on this?

No, currently having my internship so I don't have buffer on this. 🥲

**Comment by yhamacnu** [comment]
@jayzhan211 , is this still open? If - not, can I take it?

**Comment by alamb** [comment]
FWIW I am not sure there is consensus on what the correct behavior in this case is
